kernel/sched.c

```diff
@@ -1369,7 +1369,7 @@ void z_impl_k_thread_abort(struct k_thread *thread)
 int z_impl_k_thread_join(struct k_thread *thread, k_timeout_t timeout)
 {
 	k_spinlock_key_t key = k_spin_lock(&_sched_spinlock);
-	int ret = 0;
+	int ret;
```

Comment:
When go to line 1393, `ret` is not set.

Reply:
This looks OK to me.  Uninitialized variables will get caught by the compiler and result in a CI failure.  This is the opposite concern: a (frustratingly pedantic, but benign) insistence that variables **not** be initialized if there is no use before subsequent assignments.

Reply:
Thanks for your explanation.